ocaml/xapi/xapi_host.ml

```diff
@@ -1653,3 +1653,27 @@ let apply_guest_agent_config ~__context ~host =
 	let pool = Helpers.get_pool ~__context in
 	let config = Db.Pool.get_guest_agent_config ~__context ~self:pool in
 	Xapi_xenops.apply_guest_agent_config ~__context config
+
+let read_lines fi =
+	let rec loop acc =
+		let l = try Some (input_line fi) with _ -> None
+		in match l with
+			| None -> List.rev acc
+			| Some l -> loop (l :: acc)
+	in loop []
+
+let read_file file =
+	let fi = try Some (open_in file) with _ -> None in
+	match fi with
+	| None -> []
+	| Some fi ->
+		let lines = read_lines fi in
+		close_in fi ;
+		lines
+
+let patches_requiring_reboot ~__context ~host =
+  let patchs = read_file "/run/reboot-required.hfxs" in
```

Comment:
We don't need new helper functions for this: use `Stdext.Unixext.read_lines filename`.